lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -265,67 +278,205 @@ function _isAccountId(principal) {
     return (principal.length === 12 && /^\d+$/.test(principal));
 }
 
-function _checkPrincipal(requester, principal) {
-    if (principal === '*') {
-        return true;
+/**
+ * Checks if the ARN represents a root user account
+ * @param {string} arn - The ARN to check
+ * @returns {boolean} True if root user, false otherwise
+ */
+function _isRootUser(arn) {
+    if (!arn) {
+        return false;
     }
-    // User in unauthenticated (anonymous request)
-    if (requester === undefined) {
+
+    // Vault returns the following arn when the account makes requests 'arn:aws:iam::295412255825:/master/',
```

Comment:
```suggestion
    // Vault returns the following arn when the account makes requests 'arn:aws:iam::123456789012:/accountName/',
```

I don't know if this is a real id, avoid real id this is open source

And replace master with accountName to be clearer